kitsune/wiki/views.py

```diff
@@ -259,6 +253,25 @@ def document(request, document_slug, template=None, document=None):
     return response
 
 
+# Temporary view to hosts the OI UX experiments for 2 articles in the KB
+def ux_experiment_view(request, document_slug):
+    # If there is a direct request to this view
+    # and the flag is not active, redirect to the document view.
+    if (not flag_is_active(request, 'ux_experiment_1') or
```

Comment:
If this flag is set to less than 100% then this will run that percentage a 2nd time if they came from the main url. So you could in theory have someone enter the experiment from the main view, then get sent back here, and possibly have that happen multiple times. You need some way of making the user stick once a decision has been made. My recommendation would be to use a [waffle switch](https://waffle.readthedocs.io/en/stable/types/switch.html) here instead of a flag to indicate whether the entire thing is enabled or not. So here you'd check the switch, and in the main view you'd check both the switch and the flag.